### PR TITLE
Run all of the dhall package's tests on Appveyor.

### PR DESCRIPTION
**Pull request by quasicomputational**
This entails a bunch of surgery to the package set used with LTS 6.
Because I was seeing errors, I disabled allow-newer and found this
consistent package set. I also needed a newer version of `tls`, which
entailed a bunch of bumps.

### CODE REVIEW

**Comment by sjakobi** [comment]
Pinging @jneira: AFAIK we build with LTS-6 mostly to ensure compatibility with Eta.

**Comment by Gabriel439** [comment]
@quasicomputational: For the AppVeyor `doctest` failure, it looks like a locale-related issue.  I'm not sure if this will work, but try setting `LANG=en_US.UTF-8`

**Comment by jneira** [comment]
> Because I was seeing errors, I disabled allow-newer and found this consistent package set...
Wow, it is hard to did it, if cabal it is not able to help defining it

For lts-6 appveyor is asking for `mintty-0.1.2`...

**Comment by Gabriel439** [comment]
@quasicomputational: Try instead setting `LC_ALL=en_US.UTF-8`.  The reason I suggest this is that I was not able to reproduce the problem locally with `LANG`:

```
$ LANG=C cabal v1-test doctest
Preprocessing test suite 'doctest' for dhall-1.24.0..
Building test suite 'doctest' for dhall-1.24.0..
Running 1 test suites...
Test suite doctest: RUNNING...
Test suite doctest: PASS
Test suite logged to: dist/test/dhall-1.24.0-doctest.log
1 of 1 test suites (1 of 1 test cases) passed.
```

... but I was able to reproduce when setting `LC_ALL`:

```
$ LC_ALL=C cabal v1-test doctestPreprocessing test suite 'doctest' for dhall-1.24.0..
Building test suite 'doctest' for dhall-1.24.0..
Running 1 test suites...
Test suite doctest: RUNNING...
cabal: <file descriptor: 11>: hGetContents: invalid argument (invalid byte sequence)
<file descriptor: 11>: hGetContents: invalid argument (invalid byte sequence)
```

... which leads me to believe that `doctest` only cares about `LC_ALL`

**Comment by jneira** [comment]
Not sure if we will able to use `stack-lts-6.yaml` without `allow-newer:true`, it is complaining about the version of Win32....
cabal found for me a valid package set for windows (with `cabal v2-test dhall -w "\path\to\ghc-7.10.3\bin\ghc"`): https://gist.github.com/jneira/50459dcb9673fd5bab02eded678b42c9. But not sure if we can use it to get a valid `stack-lts-6.yaml`. I am currenlty trying `stack test` adding `- Win32-2.6.2.0` and `allow-newer:true`

Otoh `doctest`  fails in my local windows 7 with `chcp 850` but it works with `chcp 65001` (utf-8) although (or caused by?) its main module has `GHC.IO.Encoding.setLocaleEncoding System.IO.utf8`. So maybe calling `chcp 65001` in the test script fixes the test in appveyor

**Comment by jneira** [comment]
Anyway, why are we adding test dependencies (like tasty and so on) in `stack-lts-6.yaml` if appveyor is not testing anything for that yaml file?

**Comment by jneira** [comment]
@quasicomputational i've opened a pr with the changes to make lts-6 and doctest work

File: appveyor.yml [hunk 1]
```diff
@@ -52,7 +53,8 @@ for:
 
   # stack is not able to build test dependencies with lts-6
```
Comment (sjakobi) [inline appveyor.yml]: This comment seems outdated now. We still don't test `dhall-json` with lts-6 due to dependencies though?!

**Comment by jneira** [comment]
I am afraid we can't even build dhall-json lib or execs for lts-6.
And yeah, i think we could remove the comment thanks to this pr.

**Comment by jneira** [comment]
@quasicomputational i've opened a pr to merge with master and some minor changes: https://github.com/quasicomputational/dhall-haskell/pull/2

**Comment by jneira** [comment]
doctest are failing in appveyor due a duplicate `stack test dhall` without `chcp` in `appveyor.yaml`

**Comment by quasicomputational** [comment]
Oops - forgot to pull locally!

I've gone ahead and squashed to tidy it up. Let's see if this passes the gauntlet...

**Pull request merged**